사장님 회원가입 기능 구현 #14

shushuaka opened the pull request:
### 사장님 회원가입 (POST /api/owners/signup)

### 시나리오
1. 회원가입 요청을 받습니다.
2. 서버에서 이메일 중복 여부를 확인합니다.
3. 비밀번호는 암호화하여 데이터베이스에 저장합니다.
4. 회원가입 완료 시 성공 메시지를 반환합니다.


### 요구사항
- 이메일 중복 검사
- 이메일 암호화
- 회원가입 성공 메시지 반환

**1. DTO 생성 :** OwnerJoinReqDto, OwnerJoinRespDto 클래스 생성
**2. Entity 설정 :** Owner 엔티티 생성, @Builder 패턴 추가
**3. 컨트롤러 생성 (OwnerController) :** @RestController와 @RequestMapping("/api/owners")를 설정, signup 메서드 추가로  회원가입 요청을 처리
**4. 서비스 클래스 생성 (OwnerService) :** 사장님 회원가입 비즈니스 로직
**5. 데이터베이스 중복 체크 (OwnerRepository)**

yaewon0411 committed:
feat(domain): 도메인 확정

yaewon0411 committed:
feat(user-service): 회원가입 요청 dto 유효성 검사 추가 #3

yaewon0411 committed:
feat(password-encoder): 비밀번호 인코더 생성 #3

yaewon0411 committed:
feat(user-service): 회원가입 기능 추가 #3

yaewon0411 committed:
feat(user-controller): 회원가입 기능 추가 #3

JIMINJI1 committed:
feat(order-entity): orderStatus 기본값 설정 #5

JIMINJI1 committed:
feat(owner-entity): Enum OrderStatus 각 상태에 대한 설명 추가 #5

yaewon0411 committed:
test(user-service): 회원가입 성공 테스트 추가 #3

yaewon0411 committed:
test(user-service): 회원가입 실패 테스트 추가 #3

yaewon0411 committed:
test(user-service): 회원가입 실패 테스트 추가 #3

yaewon0411 committed:
feat(error-code): 이메일 중복 시 나가는 오류 코드 http status 수정 #3

yaewon0411 committed:
chore(application-dev.yaml): ddl-auto create으로 변경

TIP-Santa committed:
See issue #9 feat(store-controller) : 가게 생성 기능 추가

TIP-Santa committed:
See issue #9 feat(store-create-req-dto) : store-create-req-dto 추가

TIP-Santa committed:
See issue #9 feat(store-service) : 가게 생성 시 가게이름 중복검사 추가

TIP-Santa committed:
See issue #9 feat(error-code) : 가게이름 중복검사 에러코드 추가

TIP-Santa committed:
See issue #9 feat(store-repository) : 가게이름 중복검사 추가

yaewon0411 committed:
feat(login-dto): 로그인 요청 dto 추가 #8

yaewon0411 committed:
feat(user-service): 유저 로그인 기능 추가 #8

yaewon0411 committed:
feat(user-controller): 유저 로그인 기능 추가 #8

yaewon0411 committed:
feat(valid-password): 어노테이션 제약 속성 추가 #8

yaewon0411 committed:
test(user-service): 로그인 단위 테스트 추가 #8
- 로그인 성공 케이스
- 로그인 실패 테스트: 존재하지 않는 이메일, 비밀번호 불일치

yaewon0411 committed:
test(user-controller): 로그인 단위 테스트 추가 #8

yaewon0411 committed:
refactor(dto-signup): 회원가입 시 전화번호 정규 표현식 검사 추가 #8

yaewon0411 committed:
refactor(user): 빌더로 생성 시 isDeleted 필드 false로 고정되도록 수정 #8

yaewon0411 committed:
feat(argument-resolver): 메서드 파라미터로 TokenSubject 반환하는 argument resolver 추가 #13

yaewon0411 committed:
feat(user-service): 마이페이지 조회 기능 추가 #13

yaewon0411 committed:
feat(user-controller): 마이페이지 조회 기능 추가 #13

yaewon0411 committed:
feat(config): argument resolver webMVc 설정 등록 #13

yaewon0411 committed:
fix(token-subject): 생성자 인식 문제 해결 위해 ConsturctorProperties 사용 #13

yaewon0411 committed:
test(user-service): 마이페이지 조회 단위 테스트 추가 #13

yaewon0411 committed:
test(user-controller): 마이페이지 조회 단위 테스트 추가 및 통합 테스트와 분리 #13

yaewon0411 committed:
test(user-controller): 마이패이지 조회 단위 테스트 추가 #13

shushuaka committed:
feat(dto): OwnerJoinReqDto 생성 및 유효성 검사 추가 #14

- OwnerJoinReqDto 필드 정의
- 유효성 검사 어노테이션 추가

shushuaka committed:
feat(dto): OwnerJoinRespDto 생성 #14

- 회원가입 성공 시 성공 메시지를 반환

shushuaka committed:
feat(controller): 사장님 회원가입 API 추가 #14

- /api/owners/signup 추가
- 회원가입 요청 받아 OwnerService의 signup 메서드 호출
- 성공 시 OwnerJoinRespDto 반환

shushuaka committed:
feat(config): SecurityConfig 추가 및 PasswordEncoder 설정 #14

- PasswordEncoder 빈으로 BCryptPasswordEncoder 등록
- 비밀번호 암호화를 위한 Security 설정 파일 추가

shushuaka committed:
feat(repo): 이메일 중복 검사 메서드 추가 #14

- 이메일 중복 확인을 위한 existsByEmail 메서드 추가

shushuaka committed:
feat(service): OwnerService 회원가입 로직 및 비밀번호 암호화 추가 #14

- 회원가입 시 이메일 중복 검사 및 비밀버호 암호화
- 회원가입 성공 시 OwnerJoinRspDto 반환

shushuaka committed:
feat: Owner 엔티티 및 암호화 설정 추가 #14

- build.gradle에 Spring security 의존성 추가하여 비밀번호 암호화 지원
- Owner 엔티티 필드에 빌더 패턴 추가로 객체 생성 간편화

yaewon0411 committed:
feat(user-service): 유저 주소지 등록 기능 추가 #15

yaewon0411 committed:
feat(user-controller): 유저 주소지 등록 기능 추가 #15

shushuaka committed:
refactor: OwnerService 수정 및 SecurityConfig 파일 삭제 #14

- 커스텀 PasswordEncoder 사용으로 OwnerService에서 Spring Security의 PasswordEncoder 의존성 제거
- 공통 PasswordEncoder로 변경하여 SecurityConfig 파일 삭제

yaewon0411 committed:
test(user-service): 주소지 등록 단위 테스트 추가 #15

yaewon0411 committed:
chore(gitignore): yaml gitignore #15

yaewon0411 committed:
test(user-controller): 주소지 등록 단위 테스트 추가 #15

shushuaka committed:
feat(dto): OwnerLoginReqDto 생성 #16

- 사장님 로그인 요청 데이터용 DTO 생성
- 이메일과 비밀번호 필드 추가

shushuaka committed:
feat(dto): OwnerLoginRespDto 생성 #16

- 로그인 성공 시 반환할 데이터용 DTO 생성
- 성공 메시지와 JWT 토큰 필드 추가

shushuaka committed:
feat(service): OwnerService에 로그인 메서드 추가 #16

- 로그인 로직을 처리하는 login 메서드 추가
- 이메일과 비밀번호 검증 후 JWT 토큰 반환

shushuaka committed:
feat(repo): OwnerRepository에 findByEmail 메서드 추가 #16

- Optional<Owner> 타입으로 이메일 조회 기능 추가

shushuaka committed:
feat(controller): OwnerController에 로그인 API 추가 #16

- 사장님 로그인 요청을 처리할 POST /api/owners/login API 추가
- 로그인 요청 처리 후 OwnerLoginRespDto 반환

TIP-Santa reviewed (approved):
확인했습니다.

yaewon0411 commented on src/main/java/com/my/foody/domain/owner/service/OwnerService.java:
```diff
@@ -1,4 +1,71 @@
+    private final OwnerRepository ownerRepository;
+    private final JwtProvider jwtProvider;
+
+    private static final String SUCCESS_MESSAGE = "회원가입 완료 되었습니다.";
+    private static final String LOGIN_SUCCESS_MESSAGE = "로그인 성공";
```
이 메시지는 각 DTO 내에서 갖고 있어야 할 거 같습니당

yaewon0411 commented on src/main/java/com/my/foody/domain/owner/service/OwnerService.java:
```diff
@@ -1,4 +1,71 @@
+    // 이메일 중복 확인
+    private void validateEmailUniqueness(String email) {
+        if (ownerRepository.existsByEmail(email)) {
+            throw new IllegalArgumentException("이미 존재하는 이메일입니다.");
```
저희 서비스의 BusinessException을 사용해주세요

yaewon0411 commented on src/main/java/com/my/foody/domain/owner/service/OwnerService.java:
```diff
@@ -1,4 +1,71 @@
+    public OwnerLoginRespDto login(OwnerLoginReqDto reqDto) {
+        // 1. 이메일로 Owner 조회
+        Owner owner = ownerRepository.findByEmail(reqDto.getEmail())
+                .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 이메일입니다."));
```
이 부분도 BusinessException으로 나가야 합니다

yaewon0411 commented on src/main/java/com/my/foody/domain/owner/controller/OwnerController.java:
```diff
@@ -1,4 +1,38 @@
 public class OwnerController {
+    private final OwnerService ownerService;
+
+    @Autowired
+    public OwnerController(OwnerService ownerService) {
+        this.ownerService = ownerService;
+    }
```
Autowired로 주입하면 테스트 시 mock 주입이 용이하지 않습니다 따라서 모든 클래스에서 @RequiredArgsConstructor를 통한 빈 주입을 제안합니다

yaewon0411 commented on src/main/java/com/my/foody/domain/owner/controller/OwnerController.java:
```diff
@@ -1,4 +1,38 @@
+    }
+
+    @PostMapping("/signup")
+    public ResponseEntity<OwnerJoinRespDto> signup(@RequestBody OwnerJoinReqDto ownerJoinReqDto) {
+        OwnerJoinRespDto responseMessage = ownerService.signup(ownerJoinReqDto);
+        return new ResponseEntity<>(responseMessage, HttpStatus.OK);
+    }
+
+    @PostMapping("/login")
+    public ResponseEntity<OwnerLoginRespDto> login(@RequestBody OwnerLoginReqDto ownerLoginReqDto) {
+        OwnerLoginRespDto responseMessage = ownerService.login(ownerLoginReqDto);
+        return new ResponseEntity<>(responseMessage, HttpStatus.OK);
+    }
```
공통 응답 포맷을 내보내기로 결정했었습니다!  ResponseEntity 내에서 ApiResult로 한번 감싸서 내보내 주세요

yaewon0411 commented on src/main/java/com/my/foody/domain/owner/dto/req/OwnerLoginReqDto.java:
```diff
@@ -0,0 +1,16 @@
+@Getter
+public class OwnerLoginReqDto {
+
+    @Email
+    @NotBlank
+    private String email;
+
+    @NotBlank
+    private String password;
+}
```
유효성 검사 실패에 따른 메시지도 기재 바랍니다
특히 비밀번호는 정규 표현식이나 별도의 커스텀 Validator를 통해 검증해야 합니다

yaewon0411 commented on src/main/java/com/my/foody/domain/owner/controller/OwnerController.java:
```diff
@@ -1,4 +1,38 @@
+    }
+
+    @PostMapping("/signup")
+    public ResponseEntity<OwnerJoinRespDto> signup(@RequestBody OwnerJoinReqDto ownerJoinReqDto) {
+        OwnerJoinRespDto responseMessage = ownerService.signup(ownerJoinReqDto);
+        return new ResponseEntity<>(responseMessage, HttpStatus.OK);
+    }
+
+    @PostMapping("/login")
+    public ResponseEntity<OwnerLoginRespDto> login(@RequestBody OwnerLoginReqDto ownerLoginReqDto) {
```
RequestBody로 받는 객체에 @Valid가 달려있지 않으면 유효성 검사가 진행되지 않습니다 기재 부탁드려요

yaewon0411 commented on src/main/java/com/my/foody/domain/owner/dto/req/OwnerJoinReqDto.java:
```diff
@@ -1,4 +1,31 @@
+@Getter
 public class OwnerJoinReqDto {
+
+    @NotBlank
+    @Size(max = 100)
+    private String name;
+
+    @Size(max = 15)
+    private String contact;
+
+    @Email
+    @NotBlank
+    private String email;
+
+    @NotBlank
+    @Size(max = 60)
+    private String password;
+
+    @NotNull
+    private Boolean isDeleted;
```
isDeleted는 클라이언트에게 받으면 안되는 필드이기 때문에 내부에서 false로 값을 고정하는 방식을 제안합니다
각 필드에 대한 유효성 메시지를 달아주세요

그리고 비밀번호는 인코딩 했을 때의 길이가 60자이기 때문에 클라이언트에서 max=60자로 요청을 받으면 인코딩 시 db에서 제한한 길이를 넘어섭니다

여기서도 비밀번호 정규표현식을 사용해주실 때 길이를 16자 이하 정도로 길이 제약을 걸어주세요

yaewon0411 reviewed (changes requested):
테스트 코드가 안보입니다ㅠ 테스트 진행해주시기 바랍니다
현재 dev 내용도 반영해주세요!
https://github.com/A2jo/Foody/issues/2

shushuaka committed:
feat(dto): 마이페이지 조회 응답 DTO 생성 #21

- 사장님 마이페이지 조회 시 반환할 OwnerMyPageRespDto 추가
- ID, 이름, 연락처, 이메일 정보를 포함

shushuaka committed:
feat(service): OwnerService에 마이페이지 조회 메서드 추가 #21

- 사장님의 ID를 통해 마이페이지 정보를 조회하는 getMyPage 메서드 추가
- 조회된 정보를 OwnerMyPageRespDto로 반환
- 사장님 정보가 없을 경우 예외 처리

shushuaka committed:
feat(controller): OwnerController에 마이페이지 조회 API 추가 #21

- 사장님 마이페이지 조회 요청을 처리할 GET /api/owners/mypage API 추가
- ownerId로 사장님 정보를 조회하여 OwnerMyPageRespDto로 응답

shushuaka committed:
refactor: 사장님 회원가입 기능 구현 리팩토링 및 예외 처리 추가 #14

- OwnerController 응답을 ApiResult로 통일하여 공통 포맷 적용
- OwnerService의 예외 처리에 BusinessException과 ErrorCode 사용
- OwnerJoinRespDto와 OwnerLoginRespDto에 기본 메시지 필드 추가
- OwnerLoginReqDto에 유효성 검증 메시지 추가 (이메일 및 비밀번호 형식)
- BusinessException, ServerException, CustomJwtException에 로깅 기능 추가

gbognon25 reviewed (approved):
확인했습니다.

JIMINJI1 reviewed (approved):
확인했습니다
